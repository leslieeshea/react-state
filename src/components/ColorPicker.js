import React, { PureComponent } from 'react';
import styles from './ColorPicker.css';

export default class ColorPicker extends PureComponent {
  state = {
    color: ''
  };

  selectColor = color => {
    this.setState({ color });
  }

  render() {
    const { color } = this.state;

    return (
      <section className={styles.ColorPicker}>
        <button className={styles.red} onClick={this.selectColor.bind(null, 'red')}>Red</button>
        <button className={styles.yellow} onClick={() => this.selectColor('yellow')}>Yellow</button>
        <button className={styles.blue} onClick={() => this.selectColor('blue')}>Blue</button>
        <div className={styles[color]}></div>
      </section>
    );
  }
}
