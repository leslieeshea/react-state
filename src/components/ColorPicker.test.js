import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
