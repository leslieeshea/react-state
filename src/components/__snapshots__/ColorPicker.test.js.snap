// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders a ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
  <div
    className=""
  />
</section>
`;
